moirae/estimators/online/dual.py

```diff
@@ -0,0 +1,172 @@
+            initial_inputs: initial input quantities
+            covariance_transient: specifies the raw (un-normalized) covariance of the transient state; it is not used if
+                covariance_joint was provided
+            covariance_asoh: specifies the raw (un-normalized) covariance of the A-SOH; it is not used if
+                covariance_joint was provided
+            inputs_uncertainty: uncertainty matrix of the inputs; if not provided, assumes inputs are exact
+            transient_covariance_process_noise: process noise for transient update; only used if
+                joint_covariance_process_noise was not provided
+            asoh_covariance_process_noise: process noise for A-SOH update; only used if joint_covariance_process_noise
+                was not provided
+            covariance_sensor_noise: sensor noise for outputs; if denoising is applied, must match the proper
+                dimensionality
+            filter_args: dictionary where keys must be either 'transient' or 'asoh', and values must be a dictionary in
+                which keys are keyword arguments for the UKFs (such as `alpha_param`)
+        """
+        # Assemble the wrappers
+        cell_wrapper = CellModelWrapper(cell_model=cell_model,
+                                        asoh=initial_asoh,
+                                        transients=initial_transients,
+                                        inputs=initial_inputs)
+        asoh_wrapper = DegradationModelWrapper(cell_model=cell_model,
+                                               asoh=initial_asoh,
+                                               transients=initial_transients,
+                                               inputs=initial_inputs)
+
+        # Prepare multivariate Gaussians
+        transients_hidden = convert_vals_model_to_filter(model_quantities=initial_transients,
+                                                         uncertainty_matrix=covariance_transient)
+        asoh_hidden = MultivariateGaussian(mean=initial_asoh.get_parameters().flatten(),
+                                           covariance=covariance_asoh)
+        initial_controls = convert_vals_model_to_filter(model_quantities=initial_inputs)
+
+        # Initialize filters
+        trans_filter = UKF(model=cell_wrapper,
+                           initial_hidden=transients_hidden,
+                           initial_controls=initial_controls,
+                           covariance_process_noise=transient_covariance_process_noise,
+                           covariance_sensor_noise=covariance_sensor_noise)
+        asoh_filter = UKF(model=asoh_wrapper,
+                          initial_hidden=asoh_hidden,
+                          initial_controls=initial_controls,
+                          covariance_process_noise=asoh_covariance_process_noise,
+                          covariance_sensor_noise=covariance_sensor_noise)
+
+        if filter_args is not None:
```

Comment:
Having to use `setattr` worries me a bit. It's verbose and, I'm guessing, resulted you in changing the UKF constructor to have it handle receiving the tuning parameters after-the-fact. Could you instead add this to the definitions of the filters?

```python
trans_filter = UKF(..., **filter_args.get('transient', {})
```
This will retrieve the values of the transient parameters, if available, and pass them as kwargs to the UKF initializer.

You'll also need to ensure `transient_args` is not None.

```python
if transient_args is None:
     transient_args = {}
```

Reply:
Another route could be to make delayed delivery of some arguments to a filter something always supported. I could see that being useful as we build larger assemblies of filters

Reply:
Mind you, I'd suggest we follow my first proposal now. Messing with the filter seems like it's unnecessary now and, therefore, out of scope for this PR

Reply:
Came up with a way to do the first suggestion, but would like you to look it over just to make sure @WardLT.

Reply:
Nice! I hadn't seen NotRequired before. It makes a lot of sense to do it that way